### ISSUE TITLE
Include hplip for full print and scan support for HP printers

### ISSUE THREAD

**Issue by pceiley** [clearlinux/distribution#383]
**Is your feature request related to a problem? Please describe.**
I can't currently print/scan to/from my HP printer

**Describe the solution you'd like**
Include hplip package in Clear Linux

**Describe alternatives you've considered**
None.

**Additional context**
[https://developers.hp.com/hp-linux-imaging-and-printing/gethplip](url)
This is included in all major linux distributions.

**Comment by ahkok** [comment]
This is in progress of being added.

**Comment by pceiley** [comment]
Thanks for including this, but I can't get it to work.
```
$ hp-setup
warning: python-dbus not installed.
error: Unable to locate models.dat file
```

Running hpcheck gives:
```
/bin/sh: lsb_release: command not found
cat: /etc/issue: No such file or directory
warning:  distro is not found in AUTH_TYPES
/warning: Failed to get the distro information.
Traceback (most recent call last):
  File "/usr/bin/hp-check", line 861, in <module>
    dep.core.init()
  File "/usr/share/hplip/installer/core_install.py", line 494, in init
    self.distro_name = self.distros_index[self.distro]
KeyError: 0
```

**Comment by pceiley** [comment]
Anything I can do to help with this one? It’s the no. 1 blocker for me migrating to Clear so keen to get it resolved.

**Comment by pceiley** [comment]
Interestingly, doing an strace on hp-setup shows that it's looking for the models.dat file in the current working directory. So if I run it as root in /root then it does:
```
getcwd("/root", 1024)
stat("/root/data/models/models.dat", 0x.....) = -1 ENOENT (No such file or directory)
```
it should be starting in /usr/share/hplip.

**Comment by pceiley** [comment]
I'm not sure if this is even being checked, but for the sake of reporting findings:

It seems hplip relies on /etc/hp/hplip.conf, which is not present in CL. Creating one helps, but I still get python errors when running hp-setup:

```
# hp-setup -i 192.168.1.11

HP Linux Imaging and Printing System (ver. 3.18.6)
Printer/Fax Setup Utility ver. 9.0

Copyright (c) 2001-15 HP Development Company, LP
This software comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to distribute it
under certain conditions. See COPYING file for more details.

(Note: Defaults for each question are maked with a '*'. Press <enter> to accept the default.)


Setting up device: hp:/net/OfficeJet_Pro_6970?ip=192.168.1.11


Traceback (most recent call last):
  File "/usr/bin/hp-setup", line 426, in <module>
    ppds = cups.getSystemPPDs()
  File "/usr/share/hplip/prnt/cups.py", line 329, in getSystemPPDs
    path = os.path.join(cups_ppd_path, ppd)
  File "/usr/lib/python3.7/posixpath.py", line 80, in join
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType

```

**Comment by anselmolsm** [comment]
Hi @pceiley. Thanks for the comments. I see two bugs here (perhaps hidden in this issue about packaging hplip?).

- [ ] hplip seems to be looking for files at wrong places. Maybe the configure step needs adjustments.
- [ ] stateless config file (generated during the build) is not installed. Also, scripts currently look for the default config file at /etc/hp/hplip.conf only.

**Comment by pceiley** [comment]
Hi @anselmolsm. Thanks for the response!

I think the two bugs you've listed are actually the same issue. Once the config file is in place with the correct locations specified, it resolves that particular issue.

The second bug is that Python throws a TypeError when trying to execute hp-setup. I haven't had time to dig too deeply into this one.

Perhaps you're right and I should log these under a new issue? I just assumed they were teething issues having recently been packaged / included in Clear.

**Comment by anselmolsm** [comment]
@pceiley you're right, they are related. hplip scripts rely on paths defined in hplip.conf.

**Comment by gmagic** [comment]
Here's in fact one more issue.

hplip apps like (hp-toolbox, hp-print, etc.) need PyQt5 package as a minimum. Just try to execute any app of hplip (hp-*) in console and you'll see.

By the way: i could not setup network printer via Gnome - Settings - Printers - Add... the system just doesn't find it. I could install my network HP LaserJet MFP having used "hp-setup -i". And when i thought i won this struggle (mfp have been added to my printer list) i met anther issue: it just stops every printing job (wps office (flatpak) or evince. And "simple scanning" app can see scanner but can't do any scan job.

**Comment by reiki33** [comment]
hp-setup fails in QtCore.so with undefined symbol.

Is the Syntax warning responsible for the QtCore.so failure?
sudo swupd verify --picky successfully verified version 32190.

hp-setup
/usr/share/hplip/base/utils.py:2061: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if weburl is "" or weburl is None:

HP Linux Imaging and Printing System (ver. 3.19.8)
Printer/Fax Setup Utility ver. 9.0

Copyright (c) 2001-18 HP Development Company, LP
This software comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to distribute it
under certain conditions. See COPYING file for more details.

error: /usr/lib/python3.8/site-packages/PyQt5/QtCore.so: undefined symbol: _ZN8qfloat1610shifttableE, version Qt_5

**Comment by bwarden** [comment]
The Qt5 issues should be resolved in release 32220. If they persist, or for any other problems with hplip, please open a new issue.
